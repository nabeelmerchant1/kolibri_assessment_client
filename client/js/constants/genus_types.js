import _    from 'lodash';

export const types = {
  item:{
    audioUpload: 'item-genus-type%3Aqti-upload-interaction-audio%40ODL.MIT.EDU',
    dragAndDrop: 'item-genus-type%3Adrag-and-drop%40ODL.MIT.EDU',
    fileUpload: 'item-genus-type%3Aqti-upload-interaction-generic%40ODL.MIT.EDU',
    imageSequence: 'item-genus-type%3Aqti-order-interaction-object-manipulation%40ODL.MIT.EDU',
    movableFillBlank: 'item-genus-type%3Aqti-inline-choice-interaction-mw-fill-in-the-blank%40ODL.MIT.EDU',
    movableWordSandbox: 'item-genus-type%3Aqti-order-interaction-mw-sandbox%40ODL.MIT.EDU',
    movableWordSentence: 'item-genus-type%3Aqti-order-interaction-mw-sentence%40ODL.MIT.EDU',
    multipleAnswer: 'item-genus-type%3Aqti-choice-interaction-multi-select%40ODL.MIT.EDU',
    multipleChoice: 'item-genus-type%3Aqti-choice-interaction%40ODL.MIT.EDU',
    multipleReflection: 'item-genus-type%3Aqti-choice-interaction-multi-select-survey%40ODL.MIT.EDU',
    reflection: 'item-genus-type%3Aqti-choice-interaction-survey%40ODL.MIT.EDU',
    shortAnswer: 'item-genus-type%3Aqti-extended-text-interaction%40ODL.MIT.EDU',
  },
  question: {
    audioUpload: 'question-type%3Aqti-upload-interaction-audio%40ODL.MIT.EDU',
    dragAndDrop: 'question-type%3Adrag-and-drop%40ODL.MIT.EDU',
    fileUpload: 'question-type%3Aqti-upload-interaction-generic%40ODL.MIT.EDU',
    imageSequence: 'question-type%3Aqti-order-interaction-object-manipulation%40ODL.MIT.EDU',
    movableFillBlank: 'question-type%3Aqti-inline-choice-interaction-mw-fill-in-the-blank%40ODL.MIT.EDU',
    movableWordSandbox: 'question-type%3Aqti-order-interaction-mw-sandbox%40ODL.MIT.EDU',
    movableWordSentence: 'question-type%3Aqti-order-interaction-mw-sentence%40ODL.MIT.EDU',
    multipleAnswer: 'question-type%3Aqti-choice-interaction-multi-select%40ODL.MIT.EDU',
    multipleChoice: 'question-type%3Aqti-choice-interaction%40ODL.MIT.EDU',
    multipleReflection: 'question-type%3Aqti-choice-interaction-multi-select-survey%40ODL.MIT.EDU',
    reflection: 'question-type%3Aqti-choice-interaction-survey%40ODL.MIT.EDU',
    shortAnswer: 'question-type%3Aqti-extended-text-interaction%40ODL.MIT.EDU',
  },
  answer: {
    file: 'answer-record-type%3Afiles-submission%40ODL.MIT.EDU', // for any movable word sandbox, audio record tool, and generic file submission
    fillInTheBlank: 'answer-record-type%3Ainline-choice-answer%40ODL.MIT.EDU',
    multipleAnswer: 'answer-record-type%3Amulti-choice-answer%40ODL.MIT.EDU',
    multipleChoice: 'answer-record-type%3Amulti-choice-answer%40ODL.MIT.EDU', // for multiple choice, reflection, movable words, image sequence
    rightAnswer: 'answer-type%3Aright-answer%40ODL.MIT.EDU',
    shortAnswer: 'answer-record-type%3Ashort-text-answer%40ODL.MIT.EDU',
    wrongAnswer: 'answer-type%3Awrong-answer%40ODL.MIT.EDU',
  },
  assets: {
    image: {
      png: 'asset-content-genus-type%3Apng%40iana.org',
      jpg: 'asset-content-genus-type%3Ajpg%40iana.org',
      JPG: 'asset-content-genus-type%3AJPG%40ODL.MIT.EDU',
      svg: 'asset-content-genus-type%3Asvg%40iana.org',
    },
    audio: {
      mp3:'asset-content-genus-type%3Amp3%40iana.org'
    },
    video: {
      m4a: 'asset-content-genus-type%3Am4a%40ODL.MIT.EDU',
      wav: 'asset-content-genus-type%3Awav%40ODL.MIT.EDU',
      mp4: 'asset-content-genus-type%3Amp4%40ODL.MIT.EDU'
    },
    altText: {
      altText: 'asset-content-genus-type%3Aalt-text%40ODL.MIT.EDU',
    },
    description: {
      description: 'asset-content-genus-type%3Amedia-description%40ODL.MIT.EDU',
      oldDesc: 'asset-content-genus-type%3AmediaDescription%40ODL.MIT.EDU',
    },
    license: {
      license: 'someJunkyId',
    },
    copyright: {
      copyright: 'anotherJunkyId'
    },
    vtt: {
      vtt: 'asset-content-genus-type%3Avtt%40ODL.MIT.EDU',
    },
    transcript: {
      transcript: 'asset-content-genus-type%3Atranscript%40ODL.MIT.EDU',
    }
  },
  target: {
    drop: 'drop.behavior%3Adrop%40ODL.MIT.EDU',
    reject: 'drop.behavior%3Areject%40ODL.MIT.EDU',
  },
  zone: {
    drop: 'drop.behavior%3Adrop%40ODL.MIT.EDU',
    snap: 'drop.behavior%3Asnap%40ODL.MIT.EDU',
    rectangle: 'osid.mapping.SpatialUnit%3Arectangle%40ODL.MIT.EDU',
  },
  default: 'GenusType%3ADEFAULT%40DLKIT.MIT.EDU',
};

export function getQbankType(type) {
  return _.findKey(types.item, genusType => type === genusType)
    || _.findKey(types.question, genusType => type === genusType)
    || _.findKey(types.answer, genusType => type === genusType)
    || _.findKey(types.target, genusType => type === genusType)
    || _.findKey(types.zone, genusType => type === genusType)
    || null;
}

export function getQbankMediaType(uglyType) {
  return _.findKey(types.assets, mediaType => _.findKey(mediaType, mType => uglyType === mType) || null) || null;
}

export default types;
