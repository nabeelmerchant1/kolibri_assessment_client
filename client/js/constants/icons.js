import React from 'react';

// beginWrap is the arrow that goes at the end of the movable word chain line when
// wrapping to multiple lines.
export const beginWrap = <svg xmlns="http://www.w3.org/2000/svg" width="48" height="48" viewBox="0 0 48 48">
  <path d="M14.83 16.42l9.17 9.17 9.17-9.17 2.83 2.83-12 12-12-12z"/>
</svg>

// endWrap is the arrow that goes at the beginning of the movable word chain line
// on the line after beginWrapwrapping to multiple lines.
export const endWrap = <svg xmlns="http://www.w3.org/2000/svg" width="48" height="48" viewBox="0 0 48 48">
    <path d="M14.83 16.42l9.17 9.17 9.17-9.17 2.83 2.83-12 12-12-12z"/>
</svg>
