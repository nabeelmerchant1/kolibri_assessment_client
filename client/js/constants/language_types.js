import _  from 'lodash';

export const languages = {
  languageTypeId: {
    english: '639-2%3AENG%40ISO',
    hindi: '639-2%3AHIN%40ISO',
    telugu: '639-2%3ATEL%40ISO',
  },
  formatTypeId: 'TextFormats%3APLAIN%40okapia.net',
  scriptTypeId: {
    english: '15924%3ALATN%40ISO',
    hindi: '15924%3ADEVA%40ISO',
    telugu: '15924%3ATELU%40ISO',
  }
};

export function getLanguage(langCode) {
  return _.findKey(languages.languageTypeId, language => langCode === language) || null;
}

export function getScriptTypeId(langCode) {
  return _.findKey(languages.scriptTypeId, language => langCode === language) || null;
}

export function languageFromLocale(locale) {
  switch (locale) {
    case 'en':
      return 'english';
    case 'hi':
      return 'hindi';
    case 'te':
      return 'telugu';
    default:
      throw new Error('Locale not recognized');
  }
}

export function localeFromLanguage(language) {
  switch (language) {
    case '639-2%3AENG%40ISO':
      return 'en';
    case '639-2%3AHIN%40ISO':
      return 'hi';
    case '639-2%3ATEL%40ISO':
      return 'te';
    default:
      return 'en';
  }
}

export default languages;
