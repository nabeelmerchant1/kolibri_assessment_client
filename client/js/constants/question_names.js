export default {
  audioUpload: 'Audio Record',
  fileUpload: 'File Upload',
  movableWordSandbox: 'Movable Word -- Sandbox',
  movableWordSentence: 'Movable Word -- Sentence',
  movableFillBlank: 'Movable Word -- Fill in the Blank',
  shortAnswer: 'Short Answer',
  imageSequence: 'Image Sequence',
  multipleChoice: 'Multiple Choice',
  reflection: 'Multiple Choice -- Reflection',
  multipleAnswer: 'Multiple Choice -- Multiple Select',
  multipleReflection: 'Multiple Choice -- Multiple Select & Reflection',
};
