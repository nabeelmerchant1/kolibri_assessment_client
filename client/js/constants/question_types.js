export default {
  audioUpload: 'audioUpload',
  fileUpload: 'fileUpload',
  movableWordSandbox: 'movableWordSandbox',
  movableWordSentence: 'movableWordSentence',
  movableFillBlank: 'movableFillBlank',
  shortAnswer: 'shortAnswer',
  imageSequence: 'imageSequence',
  multipleChoice: 'multipleChoice',
  reflection: 'reflection',
  multipleAnswer: 'multipleAnswer',
  multipleReflection: 'multipleReflection',
  dragAndDrop: 'dragAndDrop',
};

export const audioLimit = 240;
