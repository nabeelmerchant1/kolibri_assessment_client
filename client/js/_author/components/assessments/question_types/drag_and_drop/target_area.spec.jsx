import React          from 'react';
import { shallow }    from 'enzyme';
import TargetArea     from './target_area';
import { languages }  from '../../../../../constants/language_types';

describe('target_area component', () => {
  let props;
  let result;
  let uploadedMedia;
  let zoneNew;

  beforeEach(() => {
    uploadedMedia = false;
    zoneNew = false;
    props = {
      loadingMedia: false,
      images: {},
      question: {
        target: {},
        zones: {},
      },
      uploadMedia: () => { uploadedMedia = true; },
      setVisible: () => {},
      editZone: () => { zoneNew = true; },
      language: languages.languageTypeId.english,
    };
    result = shallow(<TargetArea {...props} />);
  });

  it('renders the component as a snapshot', () => {
    expect(result).toMatchSnapshot();
  });

  it('handles the uploadMedia prop call', () => {
    expect(uploadedMedia).toBeFalsy();
    result.instance().replaceImage();
    expect(uploadedMedia).toBeTruthy();
  });

  it('handles the newZone prop call', () => {
    expect(zoneNew).toBeFalsy();
    result.instance().addByRegion();
    expect(zoneNew).toBeTruthy();
  });
});
