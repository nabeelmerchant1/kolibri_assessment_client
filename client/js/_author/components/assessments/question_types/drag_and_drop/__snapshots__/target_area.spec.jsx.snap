// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`target_area component renders the component as a snapshot 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="au-c-drag-and-drop__target-area"
>
    <targetMenu
        addByRegion={[Function]}
        addMenu={null}
        hasTarget={false}
        id={undefined}
        openModal={[Function]}
        toggleAdd={[Function]}
        toggleVisible={[Function]}
        visibleZones={undefined}
    />
    <TargetZone
        editZone={[Function]}
        language="639-2%3AENG%40ISO"
        openModal={[Function]}
        target={Object {}}
        zones={Object {}}
    />
    <EditorUploadModal
        closeModal={[Function]}
        insertMedia={[Function]}
        isOpen={false}
        language="639-2%3AENG%40ISO"
        loading={false}
        localizeStrings={[Function]}
        media={Object {}}
        mediaType="img"
        uploadOnly={false}
    />
</div>,
  "nodes": Array [
    <div
      className="au-c-drag-and-drop__target-area"
>
      <targetMenu
            addByRegion={[Function]}
            addMenu={null}
            hasTarget={false}
            id={undefined}
            openModal={[Function]}
            toggleAdd={[Function]}
            toggleVisible={[Function]}
            visibleZones={undefined}
      />
      <TargetZone
            editZone={[Function]}
            language="639-2%3AENG%40ISO"
            openModal={[Function]}
            target={Object {}}
            zones={Object {}}
      />
      <EditorUploadModal
            closeModal={[Function]}
            insertMedia={[Function]}
            isOpen={false}
            language="639-2%3AENG%40ISO"
            loading={false}
            localizeStrings={[Function]}
            media={Object {}}
            mediaType="img"
            uploadOnly={false}
      />
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <TargetArea
        editZone={[Function]}
        images={Object {}}
        language="639-2%3AENG%40ISO"
        loadingMedia={false}
        question={
                Object {
                        "target": Object {},
                        "zones": Object {},
                      }
        }
        setVisible={[Function]}
        uploadMedia={[Function]}
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": TargetArea {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "editZone": [Function],
          "images": Object {},
          "language": "639-2%3AENG%40ISO",
          "loadingMedia": false,
          "question": Object {
            "target": Object {},
            "zones": Object {},
          },
          "setVisible": [Function],
          "uploadMedia": [Function],
        },
        "refs": Object {},
        "state": Object {
          "addMenu": null,
          "addType": null,
          "showModal": false,
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="au-c-drag-and-drop__target-area"
>
          <targetMenu
                    addByRegion={[Function]}
                    addMenu={null}
                    hasTarget={false}
                    id={undefined}
                    openModal={[Function]}
                    toggleAdd={[Function]}
                    toggleVisible={[Function]}
                    visibleZones={undefined}
          />
          <TargetZone
                    editZone={[Function]}
                    language="639-2%3AENG%40ISO"
                    openModal={[Function]}
                    target={Object {}}
                    zones={Object {}}
          />
          <EditorUploadModal
                    closeModal={[Function]}
                    insertMedia={[Function]}
                    isOpen={false}
                    language="639-2%3AENG%40ISO"
                    loading={false}
                    localizeStrings={[Function]}
                    media={Object {}}
                    mediaType="img"
                    uploadOnly={false}
          />
</div>,
        "_debugID": 2,
        "_renderedOutput": <div
          className="au-c-drag-and-drop__target-area"
>
          <targetMenu
                    addByRegion={[Function]}
                    addMenu={null}
                    hasTarget={false}
                    id={undefined}
                    openModal={[Function]}
                    toggleAdd={[Function]}
                    toggleVisible={[Function]}
                    visibleZones={undefined}
          />
          <TargetZone
                    editZone={[Function]}
                    language="639-2%3AENG%40ISO"
                    openModal={[Function]}
                    target={Object {}}
                    zones={Object {}}
          />
          <EditorUploadModal
                    closeModal={[Function]}
                    insertMedia={[Function]}
                    isOpen={false}
                    language="639-2%3AENG%40ISO"
                    loading={false}
                    localizeStrings={[Function]}
                    media={Object {}}
                    mediaType="img"
                    uploadOnly={false}
          />
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <TargetArea
    editZone={[Function]}
    images={Object {}}
    language="639-2%3AENG%40ISO"
    loadingMedia={false}
    question={
        Object {
            "target": Object {},
            "zones": Object {},
          }
    }
    setVisible={[Function]}
    uploadMedia={[Function]}
/>,
}
`;
